src/main/java/org/sagebionetworks/EvaluationUtils.java

```diff
@@ -142,6 +168,84 @@ public static boolean isBooleanAnnotationPresentAndTrue(SubmissionStatus status,
 		return false;
 	}
 
+	public static void setAnnotationStringV2(SubmissionStatus status, String key, String value, boolean isPrivate) {
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotations = status.getSubmissionAnnotations();
+		if (annotations == null) {
+			annotations = new org.sagebionetworks.repo.model.annotation.v2.Annotations();
+			status.setSubmissionAnnotations(annotations);
+		}
+		if (annotations.getAnnotations() == null) {
+			annotations.setAnnotations(new HashMap<String, AnnotationsValue>());
+		}
+
+		if (annotations.getAnnotations().containsKey(key)) {
```

Comment:
This is a setter for the value, you can probably replace even if it exists already (see the previous logic in setAnnotation methods.